src/shared/lib/FisherExactTestCalculator.ts

```diff
@@ -0,0 +1,35 @@
+function getPValue(a: number, b: number, c: number, d: number, f: number[]): number {
+
+    const n: number = a + b + c + d;
+    const p: number = (f[a + b] + f[c + d] + f[a + c] + f[b + d]) - (f[a] + f[b] + f[c] + f[d] + f[n]);
+    return Math.exp(p);
+}
+
+export function getCumulativePValue(a: number, b: number, c: number, d: number): number {
+
+    let min: number, i: number;
+    const n: number = a + b + c + d;
+    let p: number = 0;
+    let f: number[] = new Array(n + 1);
```

Comment:
seems like f is log factorial, could you write that out, in the argument names too?